tiledb/sm/array_schema/dimension.h

```diff
@@ -195,13 +195,28 @@ class Dimension {
   static bool oob(
       const Dimension* dim, const void* coord, std::string* err_msg);
 
+  /** Return true if r1 is fully covered by r2. */
+  bool covered(const Range& r1, const Range& r2) const;
+
+  /** Return true if r1 is fully covered by r2. */
+  template <class T>
+  static bool covered(const Dimension* dim, const Range& r1, const Range& r2);
+
   /** Return true if the input 1D ranges overlap. */
   bool overlap(const Range& r1, const Range& r2) const;
 
   /** Return true if the input 1D ranges overlap. */
   template <class T>
   static bool overlap(const Dimension* dim, const Range& r1, const Range& r2);
 
+  /** Return ratio of the overalp of the two input 1D ranges over `r2`. */
```

Comment:
s/overalp/overlap, same below.

Should we document that the minimum return value is a positive epsilon?

Reply:
It returns `0.0` if there is not overlap (see function implementation).

Reply:
Oops you're right, thanks!